google/cloud/pubsublite/internal/partition_publisher.cc

```diff
@@ -0,0 +1,353 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsublite/internal/partition_publisher.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+
+using google::cloud::pubsublite::v1::Cursor;
+using google::cloud::pubsublite::v1::InitialPublishRequest;
+using google::cloud::pubsublite::v1::MessagePublishRequest;
+using google::cloud::pubsublite::v1::PublishRequest;
+using google::cloud::pubsublite::v1::PublishResponse;
+using google::cloud::pubsublite::v1::PubSubMessage;
+
+using UnderlyingStream = std::unique_ptr<
+    AsyncStreamingReadWriteRpc<google::cloud::pubsublite::v1::PublishRequest,
+                               google::cloud::pubsublite::v1::PublishResponse>>;
+
+PartitionPublisherImpl::PartitionPublisherImpl(
+    absl::FunctionRef<std::unique_ptr<
+        ResumableAsyncStreamingReadWriteRpc<PublishRequest, PublishResponse>>(
+        StreamInitializer<PublishRequest, PublishResponse>)>
+        resumable_stream_factory,
+    BatchingSettings const& batching_settings, InitialPublishRequest const& ipr,
+    std::unique_ptr<AlarmRegistryInterface> alarm, size_t alarm_period_ms)
```

Comment:
take a AlarmRegistryInterface& alarm_registry, you never need to own an instance